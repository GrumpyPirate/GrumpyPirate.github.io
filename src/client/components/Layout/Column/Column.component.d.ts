import { ChildrenProps } from 'types';

interface ColumnProps extends ChildrenProps {
  span?: 'auto' | number;
  spanSM?: 'auto' | number;
  spanMD?: 'auto' | number;
  spanLG?: 'auto' | number;
  spanXL?: 'auto' | number;
  push?: number;
  pushSM?: number;
  pushMD?: number;
  pushLG?: number;
  pushXL?: number;
  pull?: number;
  pullSM?: number;
  pullMD?: number;
  pullLG?: number;
  pullXL?: number;
}

export { ColumnProps };
