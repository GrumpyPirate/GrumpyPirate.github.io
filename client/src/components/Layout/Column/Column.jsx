import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import classes from './Column.scss';

const Column = ({
  span,
  spanXS,
  spanSM,
  spanMD,
  spanLG,
  spanXL,
  children,
}) => (
  <div
    className={classnames(
      classes.column, {
        [classes[`column-${span}`]]: span,
        [classes[`column-xs-${spanXS}`]]: spanXS,
        [classes[`column-sm-${spanSM}`]]: spanSM,
        [classes[`column-md-${spanMD}`]]: spanMD,
        [classes[`column-lg-${spanLG}`]]: spanLG,
        [classes[`column-xl-${spanXL}`]]: spanXL,
      },
    )}
  >
    {children}
  </div>
);

Column.propTypes = {
  span: PropTypes.number,
  spanXS: PropTypes.number,
  spanSM: PropTypes.number,
  spanMD: PropTypes.number,
  spanLG: PropTypes.number,
  spanXL: PropTypes.number,
  children: PropTypes.node,
};

Column.defaultProps = {
  span: null,
  spanXS: null,
  spanSM: null,
  spanMD: null,
  spanLG: null,
  spanXL: null,
  children: null,
};

export default Column;
