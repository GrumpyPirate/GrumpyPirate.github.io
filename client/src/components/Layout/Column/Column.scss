@import 'core';

.column {
  flex: 1 1 auto;
  padding-right: rem(map-get($gutter-widths, xs) / 2);
  padding-left: rem(map-get($gutter-widths, xs) / 2);
  @each $breakpoint in map-keys($gutter-widths) {
    @if index($gutter-widths, $breakpoint map-get($gutter-widths, $breakpoint)) > 1 {
      padding-right: rem(map-get($gutter-widths, $breakpoint) / 2);
      padding-left: rem(map-get($gutter-widths, $breakpoint) / 2);
    }
  }

  // Modifiers
  // Spans
  @for $i from 1 through $grid-columns {
    &-#{$i} {
      flex: 0 0 percentage($i / $grid-columns);
      max-width: percentage($i / $grid-columns);
    }
  }

  // Spans - per breakpoint
  @each $breakpoint in map-keys($breakpoints) {
    @if index($breakpoints, $breakpoint map-get($breakpoints, $breakpoint)) == 1 {
      @for $i from 1 through $grid-columns {
        &-#{$breakpoint}-#{$i} {
          flex: 0 0 percentage($i / $grid-columns);
          max-width: percentage($i / $grid-columns);
        }
      }
    } @else {
      @include respond-up($breakpoint) {
        @for $i from 1 through $grid-columns {
          &-#{$breakpoint}-#{$i} {
            flex: 0 0 percentage($i / $grid-columns);
            max-width: percentage($i / $grid-columns);
          }
        }
      }
    }
  }
}
