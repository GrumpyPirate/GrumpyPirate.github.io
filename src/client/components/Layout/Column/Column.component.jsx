import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import classes from './Column.component.scss';

const Column = ({
  span,
  spanSM,
  spanMD,
  spanLG,
  spanXL,
  push,
  pushSM,
  pushMD,
  pushLG,
  pushXL,
  pull,
  pullSM,
  pullMD,
  pullLG,
  pullXL,
  children,
}) => (
  <div
    className={classnames(classes.column, {
      [classes[`column-${span}`]]: span,
      [classes[`column-sm-${spanSM}`]]: spanSM,
      [classes[`column-md-${spanMD}`]]: spanMD,
      [classes[`column-lg-${spanLG}`]]: spanLG,
      [classes[`column-xl-${spanXL}`]]: spanXL,
      [classes[`column-push-${push}`]]: push,
      [classes[`column-push-sm-${pushSM}`]]: pushSM,
      [classes[`column-push-md-${pushMD}`]]: pushMD,
      [classes[`column-push-lg-${pushLG}`]]: pushLG,
      [classes[`column-push-xl-${pushXL}`]]: pushXL,
      [classes[`column-push-${pull}`]]: pull,
      [classes[`column-pull-sm-${pullSM}`]]: pullSM,
      [classes[`column-pull-md-${pullMD}`]]: pullMD,
      [classes[`column-pull-lg-${pullLG}`]]: pullLG,
      [classes[`column-pull-xl-${pullXL}`]]: pullXL,
    })}
  >
    {children}
  </div>
);

const spanTypes = [PropTypes.oneOf(['auto']), PropTypes.number];

Column.propTypes = {
  span: PropTypes.oneOfType(spanTypes),
  spanSM: PropTypes.oneOfType(spanTypes),
  spanMD: PropTypes.oneOfType(spanTypes),
  spanLG: PropTypes.oneOfType(spanTypes),
  spanXL: PropTypes.oneOfType(spanTypes),
  push: PropTypes.number,
  pushSM: PropTypes.number,
  pushMD: PropTypes.number,
  pushLG: PropTypes.number,
  pushXL: PropTypes.number,
  pull: PropTypes.number,
  pullSM: PropTypes.number,
  pullMD: PropTypes.number,
  pullLG: PropTypes.number,
  pullXL: PropTypes.number,
  children: PropTypes.node.isRequired,
};

Column.defaultProps = {
  span: null,
  spanSM: null,
  spanMD: null,
  spanLG: null,
  spanXL: null,
  push: null,
  pushSM: null,
  pushMD: null,
  pushLG: null,
  pushXL: null,
  pull: null,
  pullSM: null,
  pullMD: null,
  pullLG: null,
  pullXL: null,
};

export default Column;
