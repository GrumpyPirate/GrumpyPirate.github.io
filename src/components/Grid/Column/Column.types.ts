export interface ColumnProps {
  xs?: number;
  sm?: number;
  md?: number;
  lg?: number;
  xl?: number;
  xxl?: number;
  smDown?: number;
  mdDown?: number;
  lgDown?: number;
  xlDown?: number;
  xsOnly?: number;
  smOnly?: number;
  mdOnly?: number;
  lgOnly?: number;
  xlOnly?: number;
  push?: number;
  pushXs?: number;
  pushSm?: number;
  pushMd?: number;
  pushLg?: number;
  pushXl?: number;
  pushXxl?: number;
  pull?: number;
  pullXs?: number;
  pullSm?: number;
  pullMd?: number;
  pullLg?: number;
  pullXl?: number;
  pullXxl?: number;
}
