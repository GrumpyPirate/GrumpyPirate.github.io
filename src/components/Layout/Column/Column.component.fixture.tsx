import React from 'react';

export default {
  children: <div>Test children</div>,
};
