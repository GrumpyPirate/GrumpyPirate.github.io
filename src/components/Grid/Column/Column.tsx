import { PropsWithChildren } from 'react';
import styled, { css } from 'styled-components';

import { gridColumns, gutterWidths, mediaQueries, rem } from 'styles';

import { ColumnProps } from './Column.types';

const Column = styled.div<PropsWithChildren<ColumnProps>>`
  position: relative;
  flex: 1 0 auto;
  max-width: 100%;
  min-height: 1px;
  padding-right: ${rem(gutterWidths.xs / 2)};
  padding-left: ${rem(gutterWidths.xs / 2)};

  @media ${mediaQueries.md} {
    padding-right: ${rem(gutterWidths.md / 2)};
    padding-left: ${rem(gutterWidths.md / 2)};
  }

  /* Column sizing, UP */
  ${(props) =>
    !!props.xs &&
    css`
      flex-basis: ${(props.xs / gridColumns) * 100}%;
      max-width: ${(props.xs / gridColumns) * 100}%;
    `}
  ${(props) =>
    !!props.sm &&
    css`
      @media ${mediaQueries.sm} {
        flex-basis: ${(props.sm / gridColumns) * 100}%;
        max-width: ${(props.sm / gridColumns) * 100}%;
      }
    `}
  ${(props) =>
    !!props.md &&
    css`
      @media ${mediaQueries.md} {
        flex-basis: ${(props.md / gridColumns) * 100}%;
        max-width: ${(props.md / gridColumns) * 100}%;
      }
    `}
  ${(props) =>
    !!props.lg &&
    css`
      @media ${mediaQueries.lg} {
        flex-basis: ${(props.lg / gridColumns) * 100}%;
        max-width: ${(props.lg / gridColumns) * 100}%;
      }
    `}
  ${(props) =>
    !!props.xl &&
    css`
      @media ${mediaQueries.xl} {
        flex-basis: ${(props.xl / gridColumns) * 100}%;
        max-width: ${(props.xl / gridColumns) * 100}%;
      }
    `}
  ${(props) =>
    !!props.xxl &&
    css`
      @media ${mediaQueries.xxl} {
        flex-basis: ${(props.xxl / gridColumns) * 100}%;
        max-width: ${(props.xxl / gridColumns) * 100}%;
      }
    `}

  /* Column sizing, DOWN */
  ${(props) =>
    !!props.xlDown &&
    css`
      @media ${mediaQueries.xlDown} {
        flex-basis: ${(props.xlDown / gridColumns) * 100}%;
        max-width: ${(props.xlDown / gridColumns) * 100}%;
      }
    `}
  ${(props) =>
    !!props.lgDown &&
    css`
      @media ${mediaQueries.lgDown} {
        flex-basis: ${(props.lgDown / gridColumns) * 100}%;
        max-width: ${(props.lgDown / gridColumns) * 100}%;
      }
    `}
  ${(props) =>
    !!props.mdDown &&
    css`
      @media ${mediaQueries.mdDown} {
        flex-basis: ${(props.mdDown / gridColumns) * 100}%;
        max-width: ${(props.mdDown / gridColumns) * 100}%;
      }
    `}
  ${(props) =>
    !!props.smDown &&
    css`
      @media ${mediaQueries.smDown} {
        flex-basis: ${(props.smDown / gridColumns) * 100}%;
        max-width: ${(props.smDown / gridColumns) * 100}%;
      }
    `}

  /* Column sizing, ONLY */
  ${(props) =>
    !!props.xsOnly &&
    css`
      @media ${mediaQueries.xsOnly} {
        flex-basis: ${(props.xsOnly / gridColumns) * 100}%;
        max-width: ${(props.xsOnly / gridColumns) * 100}%;
      }
    `}
  ${(props) =>
    !!props.smOnly &&
    css`
      @media ${mediaQueries.smOnly} {
        flex-basis: ${(props.smOnly / gridColumns) * 100}%;
        max-width: ${(props.smOnly / gridColumns) * 100}%;
      }
    `}
  ${(props) =>
    !!props.mdOnly &&
    css`
      @media ${mediaQueries.mdOnly} {
        flex-basis: ${(props.mdOnly / gridColumns) * 100}%;
        max-width: ${(props.mdOnly / gridColumns) * 100}%;
      }
    `}
  ${(props) =>
    !!props.lgOnly &&
    css`
      @media ${mediaQueries.lgOnly} {
        flex-basis: ${(props.lgOnly / gridColumns) * 100}%;
        max-width: ${(props.lgOnly / gridColumns) * 100}%;
      }
    `}
  ${(props) =>
    !!props.xlOnly &&
    css`
      @media ${mediaQueries.xlOnly} {
        flex-basis: ${(props.xlOnly / gridColumns) * 100}%;
        max-width: ${(props.xlOnly / gridColumns) * 100}%;
      }
    `}

  /* Column pushes */
  ${(props) =>
    !!props.push &&
    css`
      left: ${(props.push / gridColumns) * 100}%;
    `}
  ${(props) =>
    !!props.pushXs &&
    css`
      @media ${mediaQueries.xsOnly} {
        left: ${(props.pushXs / gridColumns) * 100}%;
      }
    `}
  ${(props) =>
    !!props.pushSm &&
    css`
      @media ${mediaQueries.sm} {
        left: ${(props.pushSm / gridColumns) * 100}%;
      }
    `}
  ${(props) =>
    !!props.pushMd &&
    css`
      @media ${mediaQueries.md} {
        left: ${(props.pushMd / gridColumns) * 100}%;
      }
    `}
  ${(props) =>
    !!props.pushLg &&
    css`
      @media ${mediaQueries.lg} {
        left: ${(props.pushLg / gridColumns) * 100}%;
      }
    `}
  ${(props) =>
    !!props.pushXl &&
    css`
      @media ${mediaQueries.xl} {
        left: ${(props.pushXl / gridColumns) * 100}%;
      }
    `}
  ${(props) =>
    !!props.pushXxl &&
    css`
      @media ${mediaQueries.xxl} {
        left: ${(props.pushXxl / gridColumns) * 100}%;
      }
    `}

  /* Column pulls */
  ${(props) =>
    !!props.pull &&
    css`
      left: ${(props.pull / gridColumns) * -100}%;
    `}
  ${(props) =>
    !!props.pullXs &&
    css`
      @media ${mediaQueries.xsOnly} {
        left: ${(props.pullXs / gridColumns) * -100}%;
      }
    `}
  ${(props) =>
    !!props.pullSm &&
    css`
      @media ${mediaQueries.sm} {
        left: ${(props.pullSm / gridColumns) * -100}%;
      }
    `}
  ${(props) =>
    !!props.pullMd &&
    css`
      @media ${mediaQueries.md} {
        left: ${(props.pullMd / gridColumns) * -100}%;
      }
    `}
  ${(props) =>
    !!props.pullLg &&
    css`
      @media ${mediaQueries.lg} {
        left: ${(props.pullLg / gridColumns) * -100}%;
      }
    `}
  ${(props) =>
    !!props.pullXl &&
    css`
      @media ${mediaQueries.xl} {
        left: ${(props.pullXl / gridColumns) * -100}%;
      }
    `}
  ${(props) =>
    !!props.pullXxl &&
    css`
      @media ${mediaQueries.xxl} {
        left: ${(props.pullXxl / gridColumns) * -100}%;
      }
    `}
`;

export default Column;
