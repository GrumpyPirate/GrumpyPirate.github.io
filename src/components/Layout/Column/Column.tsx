import classnames from 'classnames';
import React, { FunctionComponent } from 'react';

import classes from './Column.scss';
import { ColumnProps } from './Column.types';

const Column: FunctionComponent<ColumnProps> = ({
  span,
  spanSM,
  spanMD,
  spanLG,
  spanXL,
  push,
  pushSM,
  pushMD,
  pushLG,
  pushXL,
  pull,
  pullSM,
  pullMD,
  pullLG,
  pullXL,
  children,
}) => (
  <div
    className={classnames(classes.column, {
      [classes[`column-${span}`]]: span,
      [classes[`column-sm-${spanSM}`]]: spanSM,
      [classes[`column-md-${spanMD}`]]: spanMD,
      [classes[`column-lg-${spanLG}`]]: spanLG,
      [classes[`column-xl-${spanXL}`]]: spanXL,
      [classes[`column-push-${push}`]]: push,
      [classes[`column-push-sm-${pushSM}`]]: pushSM,
      [classes[`column-push-md-${pushMD}`]]: pushMD,
      [classes[`column-push-lg-${pushLG}`]]: pushLG,
      [classes[`column-push-xl-${pushXL}`]]: pushXL,
      [classes[`column-push-${pull}`]]: pull,
      [classes[`column-pull-sm-${pullSM}`]]: pullSM,
      [classes[`column-pull-md-${pullMD}`]]: pullMD,
      [classes[`column-pull-lg-${pullLG}`]]: pullLG,
      [classes[`column-pull-xl-${pullXL}`]]: pullXL,
    })}
  >
    {children}
  </div>
);

export default Column;
