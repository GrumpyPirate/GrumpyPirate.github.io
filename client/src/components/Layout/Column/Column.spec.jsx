import React from 'react';
import Column from './Column';

describe('<Column />', () => {
  const props = {
    children: <div>Test children</div>,
  };

  it('should exist', () => {
    expect(Column).toBeDefined();
  });

  it('should be a function', () => {
    expect(typeof Column).toBe('function');
  });

  it('should render as expected, without crashing', () => {
    const wrapper = (<Column {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
