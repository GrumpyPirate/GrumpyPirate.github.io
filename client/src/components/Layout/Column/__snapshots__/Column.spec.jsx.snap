// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Column /> should render as expected, without crashing 1`] = `
<Column
  pull={null}
  pullLG={null}
  pullMD={null}
  pullSM={null}
  pullXL={null}
  push={null}
  pushLG={null}
  pushMD={null}
  pushSM={null}
  pushXL={null}
  span={null}
  spanLG={null}
  spanMD={null}
  spanSM={null}
  spanXL={null}
>
  <div>
    Test children
  </div>
</Column>
`;
